examples/rtl_433_mqtt_hass/rtl_433_mqtt_hass.py

```diff
@@ -13,308 +13,54 @@
 from __future__ import print_function
 from __future__ import with_statement
 
+import os
+import argparse
+import ast
 import time
 import json
 import paho.mqtt.client as mqtt
 
-MQTT_HOST = "127.0.0.1"
-MQTT_PORT = 1883
-MQTT_TOPIC = "rtl_433/+/events"
-# When MQTT_USERNAME is set to None it will disable username and password for mqtt
-MQTT_USERNAME = None
-MQTT_PASSWORD = None
-DISCOVERY_PREFIX = "homeassistant"
-DISCOVERY_INTERVAL = 600  # Seconds before refreshing the discovery
+"""Parse arguments provide sane defaults"""
+parser = argparse.ArgumentParser()
+parser.add_argument("--host", help="MQTT host", default="127.0.0.1", required=False)
+parser.add_argument("--port", help="MQTT port", default=1883, type=int, required=False)
+parser.add_argument("--topic", help="MQTT topic", default="rtl_433/+/events", required=False)
+parser.add_argument("--username", help="MQTT username", default=None, required=False)
+parser.add_argument("--password", help="MQTT password", default=None, required=False)
+parser.add_argument("--prefix", help="Discovery prefix", default="homeassistant", required=False)
+parser.add_argument("--interval", help="Discovery interval", default=600, type=int, required=False)
+parser.add_argument("--mapfile", help="Mapping file", default="example_mappings.json", required=False)
+parser.add_argument("-e", help="configure from environment variables", action="store_true", required=False)
+args = parser.parse_args()
+
+"""Override Arguments with Env vars if required"""
+if args.e:
```

Comment:
Processing environment variables after command line arguments means the environment variables take precedent, so they can't be overridden with from the command line.